pkg/aws/ec2/api/wrapper.go

```diff
@@ -492,6 +503,31 @@ func (e *ec2Wrapper) getClientUsingAssumedRole(instanceRegion string, roleARN st
 
 }
 
+func (e *ec2Wrapper) createSTSClient(
+	userStsSession *session.Session,
+	client *http.Client,
+	endpoint endpoints.ResolvedEndpoint,
+	sourceAcct, sourceArn string) *sts.STS {
+
+	stsClient := sts.New(userStsSession, aws.NewConfig().WithHTTPClient(client).
+		WithEndpoint(endpoint.URL).WithMaxRetries(MaxRetries))
+
+	// if region is not specified, we fallback to the default client
+	parsedArn, err := arn.Parse(sourceArn)
+	if err == nil && parsedArn.Region != "" {
+		stsClient.Handlers.Sign.PushFront(func(s *request.Request) {
+			s.ApplyOptions(request.WithSetRequestHeaders(map[string]string{
+				SourceKey:  sourceArn,
+				AccountKey: sourceAcct,
+			}))
+		})
+	} else {
+		e.log.Info("Will use default STS client since unexpected error or cluster ARN", "Errors", err, "ParsedARN", parsedArn)
```

Comment:
Just noting, falling to default client seems like better approach than crashing.